### ISSUE TITLE
Rename Resource Descriptor class

### ISSUE THREAD

**Issue by nicholascar** [w3c/dxwg#573]
The Resource Descriptor class' name may not convey its intention well. This name needs to be reconsidered with discussion about it here.

**Comment by agbeltran** [comment]
Discussion about this name also happening in https://github.com/w3c/dxwg/issues/404

**Comment by aisaac** [comment]
For the record, for a similar notions, other proposals (my F2F3 diagram [1] and our F2F postits [2]) have used the following words, without making a final decision:
- profile description/specification/expression
- distribution/publication

Implementation may also be an option...

[1] https://docs.google.com/drawings/d/1dHkpwKwUwMgS1RqSCTPO3uOoRiY_qNk0z5bhXJlYi4Y/
[2] currently in the first page of https://docs.google.com/document/d/13hV2tJ6Kg2Hfe7e1BowY5QfCIweH9GxSCFQV1aWtOPg/

**Comment by smrgeoinfo** [comment]
also a related suggestion w3c/dxwg#529, copied here
In the Figure 1. OWL [owl2-overview] diagram of this ontology, the ResourceDescriptor class is basically an associationClass that associates a resource with the Profile in some role with some purpose.  There might be multiple representations of this resource available, and these will have different linking information (URL, Mime Type etc.). The dcat:Distribution class is defined to model exactly this; shouldn't the target of the 'artifact' predicate/role  be dcat:Distribution?  The model would look something like this:
![image](https://user-images.githubusercontent.com/513380/47945975-4a386900-dec3-11e8-86db-c9fbf59c24db.png)

**Comment by kcoyle** [comment]
I'm not convinced that the resource descriptor is necessary. If most supporting artifacts are one to one with a resource descriptor then they can be reduced to a single thing, and that thing is the supporting resource, which has a role and a physicality (MIME type). In the few cases where one provides a resource in more than one physical format (e.g. ShEx in JSON-LD and in RDF) it's not a big deal to describe it twice. Basically, this provides a level of indirection for a small number of cases that can be handled without it. (And BTW I would discourage people from keeping around the same data in multiple formats - that's a maintenance nightmare.)

I really like calling the resource a "supporting resource" and that resource would be a Thing - an artifact. 

I also do not think that we should assume that profile artifacts will be described with DCAT. In fact, we haven't so far seen any proposed metadata for describing profile artifacts beyond _role_ , _dct:format_, and _profileOf_. There's no title, no description, no license ... basically there is no metadata for the content of the artifacts themselves.

**Comment by makxdekkers** [comment]
@kcoyle In my reading, `prof:Profile` is on the same level as `dcat:Dataset` and `prof:ResourceDescriptor` is on the level of `dcat:Distribution`, with `prof:hasArtifact` being the equivalent of `dcat:downloadURL`. 
I see no reason that we could not add more properties to `prof:ResourceDescriptor`, adding things like title, description and licence to it.
I do think that we need a new name for it. I would suggest something like ImplementationResource or ProfileDistribution.

**Comment by andrea-perego** [comment]
@makxdekkers said:

> @kcoyle In my reading, prof:Profile is on the same level as dcat:Dataset and prof:ResourceDescriptor is on the level of dcat:Distribution, with prof:hasArtifact being the equivalent of dcat:downloadURL.
I see no reason that we could not add more properties to prof:ResourceDescriptor, adding things like title, description and licence to it.

+1

>I do think that we need a new name for it. I would suggest something like ImplementationResource or ProfileDistribution.

I would strongly support `prof:ProfileDistribution`.

**Comment by kcoyle** [comment]
@andrea-perego The problem with "implementation resource" is that not all of the resources are directly related to implementation, such as various explanatory documents for humans. You could argue that EVERYTHING aids implementation, but I think that many people will associate implementation with code. The problem with "profile distribution" is that distribution is well understood by DCAT users but the profile work here is not to be dependent on knowledge of DCAT. 

I still question whether an abstract layer is needed. I know that it is common to have them in OO design but this is a vocabulary, not code. As a vocabulary, it is to be at least in part human-facing. An abstract layer would need to be justified by a significant presence of one-to-many situations - that is, one resource description to many distributions. I don't think that is the case here. And as I've said above, it's generally a bad idea to have multiple versions of the same thing due to the need to keep them in sync.

Note: the one-to-many in FRBR - which some cite in relation to DCAT design - is a frequent case and thus is a viable design for library catalogs. But FRBR is not a good model for standards documents where precision and 'equivalence' matter. The "many" in FRBR can vary widely, creatively, such as translations of poetry, movies made from books, or entirely updated versions of textbooks. It's an entirely different use case. For profiles, even certain minor differences could result in a different profile, not a new distribution. If you have one profile with terms A, B, C, D and another with A, B, C, D, E you cannot use the same validation rules with both. Minor differences count.

**Comment by makxdekkers** [comment]
@kcoyle What worries me here is that you seem to question fundamental aspects of the model. If that's the case, there is no way we can publish a FPWD.
As far as understand, the profile model is quite similar to the DCAT model. I don't think that is a problem. Rather, I think it is a strength.
Personally, I see no problem with ProfileDistribution -- if the class is well defined, a reader that knows about DCAT would see the similarity, and a reader who doesn't know about DCAT would just read the definition and that's it.
Of course there are one-to-many relationships, at least the way I understand the model. In the case of DCAT-AP, https://joinup.ec.europa.eu/release/dcat-ap/12 is the `prof:Profile` and each of the things linked at the bottom of that page are instances of `prof:ProfileDistribution`. They are actually listed under a heading "Distributions"(!)
If you look at https://joinup.ec.europa.eu/rdf_entity/http_e_f_fdata_ceuropa_ceu_fw21_fb8e6a3e2_b372e_b4be3_b8cc6_b938c7a2181ce (unfortunately not a nice URI), there you have the format and a download link (prof:hasArtifact).
In my mind, your example of the two profiles X(A,B,C,D) and Y(A,B,C,D,E) are two different profiles and would be two instances of prof:Profile, with each having their own a set of ProfileDistributions, e.g. https://joinup.ec.europa.eu/release/dcat-ap/11 vs. https://joinup.ec.europa.eu/release/dcat-ap/12.

**Comment by andrea-perego** [comment]
I share @makxdekkers's point.

**Comment by kcoyle** [comment]
Again, @makxdekkers I think you are looking at this from a DCAT point of view, and what we are charged with developing is not a DCAT-dependent standard. We need to assume that many folks who use the profile guidance and terms will have no knowledge of DCAT - and mau not have the use cases that formed the DCAT model. We have removed the link to the DCAT document from the guidance document for that reason. If we want to test the one-to-many hypothesis we need examples beyond DCAT, IMO. DCAT should not be our base.

But, yes, we can publish this as a FPWD because anything can change between the FPWD and subsequent drafts. The purpose is to gather feedback. We should definitely make it a point to gather feedback from beyond the DCAT community. If people like the model as is, then that's what we'll go for. But let's look at profiles with the use cases and requirements of profiles in mind.

**Comment by makxdekkers** [comment]
@kcoyle No, I wasn't looking at this from a DCAT perspective. As I wrote, people looking at the profile ontology who had no knowledge of the DCAT approach would just look at the spec and take the definition of ProfileDistribution, and if it was described well, would just take it like that.
Did you look at my example of DCAT-AP and the way that profile is described and distributed? I honestly do not see another way to publish a profile and a set of associated files like the ones published for DCAT-AP. I would love to see your counterproposal of how it could work without ProfileDistribution objects to describe the various files associated with the profile.
I am not worried about the feedback we get on a FPWD -- I just think that we, as the group that proposes it, need to agree on the top-level model. If we don't, we should not go out for FPWD, is my opinion.

**Comment by makxdekkers** [comment]
@kcoyle Now I am confused. I had the impression that we were discussing the current model at https://w3c.github.io/dxwg/profilesont/, and in particular, the diagram at https://w3c.github.io/dxwg/profilesont/profilesont.svg, not @smrgeoinfo's model higher up in this thread.
I reacted to your comment that you were "_not convinced that the resource descriptor is necessary_". Your argument, based on the properties for Profile and ResourceDescriptor **in the current model**, seemed to imply you wanted to change the model as it is now, associating the dct:format, prof:hasArtifact and prof:Role directly with prof:Profile and removing the prof:ResourceDescriptor from the model.
If that was not your intention, then I have completely misunderstood.

**Comment by makxdekkers** [comment]
Agreed, the definition of prof:ProfileDistribution should not be the same as dcat:Distribution. It should at least be more flexible, to include the types of objects associated with DCAT-AP (listed as Distributions, i.e. PDF/DOC/ODT specs, Turte file, SHACL file, SVG/PNG/EA diagrams).

**Comment by kcoyle** [comment]
Makx, We'll just have to agree to disagree. I do not consider those files "distributions" in the DCAT sense - for the reasons I have given above - and I also don't consider them "distributions" in the natural language sense since they are parts that make up a whole, not distributions of an existing set of data. They are different objects with different data content, albeit about the same conceptual thing. Some may actually be transformations of each other (e.g. the ODT and DOCX), and I think that is a special relationship that should be noted,  but others are independently created, and many have unique content. For these reasons, I see no reason to call them distributions, which might cause confusion with dcat:Distribution. They are better described, IMO, as they are in the profiles ontology as resources or objects with roles. The objects themselves _might_ be said to have distributions, and that would cover the transformed files, like ODT/DOCX/PDF if they aren't independently created, but we need to think about that more (profilesOnt doesn't get into that kind of detail or definition). In FRBR those would be different manifestations because items must be exact copies of each other, but [FaBio](https://sparontologies.github.io/fabio/current/fabio.html) treats different serializations or file formats as different items, which also makes sense. 

I'm going to try a different diagram style, one that is more "thing"-based than the ones we have now. I'll see if that can help clarify my thinking. It would help me to see some datasets that are defined in DCAT. I can find documents, but I'm not finding a way to look at the DCAT descriptions of catalogs and datasets and distributions. That would help me a lot, thanks.

**Comment by nicholascar** [comment]
I agree with @kcoyle that we should avoid 'distribution' in any part of this classe's name as that would  give the DCAT-based impression that this class is a representation of the Profile's whole which won't be the case. We know people will use this class to link in guidance documents, validation files etc. which are only parts of the whole profile. 

In Issue https://github.com/w3c/dxwg/issues/572 we are circling around 'profile object' or other, similar, terms that indicate a part of the (Profile's) whole, rather than a distribution of the profile as a whole. I like that direction better and wonder about the general use then of `:Profile dct:hasPart :ThisClass .`. since the decinition of `dct:hasPart` is "A related resource that is included either physically or logically in the described resource.", to me, this class does represent some such object, with the additional requirement we think we have to indicate a role for that part.

If this class were called `ProfileObject` (I'm not trilled by "object" as, like "resource" it's generic) and had the properties we currenly indicate for it (a `dct:format`, `dct:conformsTo`, `prof:role`) then we might have something that's simpler (using more well-known DCT terms, fewer custom terms) and yet able to do what `ProfileDescritpr` was designed to do.

@rob-metalinkage: would use of `dct:hasPart` to indicate object of this class from a `Profile` object be problematic? Assuming this object still then lead to the artifact itself?

**Comment by kcoyle** [comment]
Note that we could also use dct:hasFormat which, in spite of its name, means "A related resource that is substantially the same as the pre-existing described resource, but in another format." So if the same resource/object/thing is available in multiple formats (like DOCX/ODT or SVG/PNG) you can indicate that they are the same content in a different serialization/format. There are different ways to fit that into the model, I've got some diagrams in progress but don't know if we should continue this here.

**Comment by nicholascar** [comment]
@kcoyle I think it reasonable to continue this here. We haven't got a better place for this really.

I know the `dct:hasFormat` property and I think what you're saying is that you could have something like this:

```
# just testing these names, not promising anything!
:Profile_X a prof:Profile ;
  prof:hasPart 
    :ProfileObject_M , 
    :ProfileObject_N , 
    :ProfileObject_Y , 
    :ProfileObject_Z .

# a guidance doc
:ProfileObject_M a prof:ProfileObject ;
  :hasRole role:Guidance ;
  dct:format <http://w3id.org/mediatype/application/pdf> ;
  dct:hasFormat :ProfileObject_Z .  # i.e. there is another 'form' of this 'ProfileObject'

# the same thing as :ProfileObject_M excpet it's a PDF, this is a Word doc
:ProfileObject_N dct:format <http://w3id.org/mediatype/application/msword> .

# a SHACL constraints document
:ProfileObject_Y a prof:ProfileObject ;
  :hasRole role:Validation ;
  dct:format <http://w3id.org/mediatype/application/msword> ;
  dct:hasFormat :ProfileObject_Z ;
  dct:conformsTo <http://www.w3.org/ns/shacl#> .  

# all the same properties as :ProfileObject_Y except for dct:format
:ProfileObject_Z dct:format <http://w3id.org/mediatype/application/ld+json> .
```
Is that what you meant? I can see that this would be a quick way to represent parts of profiles that are substantially the same with just a format difference. Here I'm using my above proposed `dct:hasPart` to indicate these 'ProfileObjects' within the overall Profile. the `hasRole` is giving all of the parts meaning, so we haven't lost anything from the original ontology, but this all seems simpler and more intuitive than Descriptors, Deistributions etc.

**Comment by makxdekkers** [comment]
I agree with calling the class prof:ProfileObject. I'd rather define a special property prof:hasProfileObject because I don't think a specification or SHACL file is a 'part of' a profile in any sense of the word.
Using dct:hasFormat  to link a ProfileObject for a DOC file to a ProfileObject for an ODT file is OK ; it's exactly what the property is for.

**Comment by kcoyle** [comment]
Here are crude versions of what I meant with dct:hasFormat. I think one of these is more or less as you coded above. I also think that the model could allow both, depending on what users need at the time.

![profilediagrams 001](https://user-images.githubusercontent.com/1564129/48717170-b3132700-ebcd-11e8-9e1e-b11f518a00a1.jpeg)
![profilediagrams 002](https://user-images.githubusercontent.com/1564129/48717171-b3132700-ebcd-11e8-9e30-f1742bbc6f90.jpeg)

**Comment by kcoyle** [comment]
I agree with @makxdekkers about "hasPart" - because there is no whole, really. And we can decide to shorten hasProfileObject to something like hasProfObj if we want, leaving the label as 'has profile object'. In fact, given that this is the prof vocabulary, it could be `prof:hasObject` which includes the necessary context. 

In the end I don't care greatly what we call it, but it's not a 'part' in my mind.

**Comment by smrgeoinfo** [comment]
is prof:hasObject different from dct:relation, other than a (implicit?) domain restriction?

**Comment by kcoyle** [comment]
My answer here is pretty much same as to 'conformsTo' - it depends on how specific you want to be. Could there be related files (and dct:relation is ANY related file) that are not profile objects? Would the addition of prof:role be enough to make that clear? I also would think about how one might search in a heterogeneous graph space in order to produce a suitable use case. (p.s. prof:hasObject might well be defined as a subclass of dct:relation.)

**Comment by nicholascar** [comment]
OK, I agree that `dct:hasPart` can be shelved as it gives a part/whole impression that we can't guarentee to support.

@smrgeoinfo: I worry that `dct:relation` just doesn't tell you much and is probably too broad, as @kcoyle suggests! Karen may be right though taht duck typing anthing indicated by `dct:relation` with a `prof:role` might work, as would a `prof:hasObject` subPropertyOf `dct:relation`.

In general, I like to reuse well-known properties and then specialise their range object with duck typing (so extra properties, not subclassing) rather than creating specialised properties with subPropertyOf, since most people don't use inferencing and won't then know to relate the specialised property back to the general property. This might be more art than science though; I have no number to back up my assumption.

@rob-metalinkage: we need to hear from you here.

**Comment by smrgeoinfo** [comment]
I like the idea of having a :role on any dct:relation.

**Comment by nicholascar** [comment]
@smrgeoinfo it's the qualified relations pattern (http://patterns.dataincubator.org/book/qualified-relation.html) all over again, something I pushed hard for in the early DCAT work (now adopted in places).

**Comment by makxdekkers** [comment]
I am not sure what would be gained by declaring `prof:hasProfObject` a sub-property of `dct:relation`. In fact, it has been argued by some people at DCMI that `dct:relation` is the implicit super-property of any predicate -- as every predicate defines a relationship between a Subject and an Object, but that tells you basically nothing.
I do not fully understand how one would assert a Role on a `dct:relation`. The basic pattern is 

`:Something dct:relation :SomethingElse`

Where would the Role go?

**Comment by makxdekkers** [comment]
@kcoyle I just wanted to react to you statement above: "_I do not consider those files [...] "distributions" in the natural language sense_". 
The case is that at least one maintainer of an application profile, https://joinup.ec.europa.eu/release/dcat-ap/12, does call those associated files 'Distributions' so I am not sure that our discussion about the use of this term is relevant. As long as the definitions of classes and properties in the ontology are clear, there should not be a problem how they are called.

**Comment by nicholascar** [comment]
What about **Profiling Resource**? 

Using "Profiling", we avoid issues to do with interpretations of *Implementation*, *Supporting* etc. as *Profiling* can just be defined under the umbrella of the definition and class for *Profile* and then this thing - the *...Resource* - is just a thing that gets the profiling done and we don't have to worry about any more defining than that.

**Comment by aisaac** [comment]
@nicholascar I'm not so enthusiastic about it, honeslty. I'd prefer `ProfileResource` as I think "profiling" may be confusing (i.e. one could understand that the resource profiles a profile). And then we're not far from `ProfileObject` already suggested.

**Comment by rob-metalinkage** [comment]
After going through the nature of ResourceDescriptors in issue w3c/dxwg#769 it feels we are in a better place to reconsider this.  I would say the shape of model is confirmed, but perhaps it is worth revisiting the name in order to make it easier to grasp the nature of things?

options:

1) leave it as ResourceDescriptor
2) change to ProfileResource 
3) something else

i think there is no issue with making a change as we dont have an installed base (i plan implement as soon as we feel we have processed this first round of feedback and stablised namings.)

so @kcoyle should we create a new issue to have a vote (or put one at the end of this long one)

**Comment by kcoyle** [comment]
We could, but w3c/dxwg#769 pretty much negates the name "ProfileResource" since that issue points out that the graph is not about the resource but about the resource role. So until we resolve that issue we can't resolve this one.

**Comment by kcoyle** [comment]
Also, w3c/dxwg#769 is far from resolved and there is still a major question about the nature of the profile resource. (Sorry for the multiple comments - it's very early and I'm rushing to catch a train. I'll stop now until I get settled and can think things out.)

**Comment by kcoyle** [comment]
Thanks, Richard. We can set up a vote on that. I do have one question about the use of dcat:distribution: What would be the URI that is the  object of :hasArtifact in that diagram? Is it the URI of the artifact itself? (That, to me, would be correct.) If it is, what is the role of dcat:accessURL in that scenario?

**Comment by smrgeoinfo** [comment]
What I was thinking is that the prof:hasArtifact/dcat:Distribution is about the specific representation (as a resource, or rdf-triple-object) of a resource describing the profile in some role (what prof:ResourceDescriptor does). The dcat:accessURL is the location where that representation can be 'GET'ed on the web.  The dcat:Distribution value for the prof:hasArtifact can be a blank node (as in the example I posted); it could be assigned an identifier if that is useful. 
I think a possible source of confusion here is that the current approach links from a profile, which I view as a resource (FRBR work), to related resource (resourceDescription) representations (hasArtifact) directly, instead of linking the profile resource to a description resource, recognizing that both of these might have one or more representations (realized as dcat:Distributions, which are resources of a different type, that conform to some specification and have some representation format). I'm fully aware of the issue that different representations of a resource can be accessed at different distribution:accessURL, or via content negotiation.  It is a profile WG issue about how to communicate the content negotiation (http accept header syntax) options in the metadata, saving clients from feeling around with HEAD requests and guessing what the response means. Using different distribution:accessURL can avoid the problem by using dct:conformsTo and dct:format for each distribution.

**Comment by kcoyle** [comment]
@smrgeoinfo The problem that I have with the DCAT distribution is that it is not properly reified - nothing in it states that the triples there are statements about the file found at the access URL. If those statements are to hold true for the access URL then the access URL should be the object of 'hasArtifact'. The other option is to explicitly reify the distribution to the file being described using "rdf:about". The third option, which I think cannot be the case, would be to require that the URI that is the object of "hasArtifact" and the subject of the thing of type "dcat:Distribution" be required to respond appropriately to content negotiation for requests for a certain file type. 

I am also assuming that under this design, two files in different formats are different distributions . This was confirmed elsewhere by @makxdekkers.

**Comment by makxdekkers** [comment]
@kcoyle You make an interesting observation about the "reification" of `dcat:Distribution`. As far as I know, all users of DCAT take for granted that the triples are indeed about the file found at access URL even if the specification does not explicitly say so.

I do, however, agree that @smrgeoinfo's diagram at https://github.com/w3c/dxwg/issues/573#issuecomment-438688720 contains one level too many. I would think that `prof:ResourceDescriptor` is on the same level as `dcat:Distribution` -- they share a lot of properties, with `prof:ResourceDescriptor` just adding type and role.

**Comment by kcoyle** [comment]
@makxdekkers Thanks, Makx. I'm sure that the DCAT users are all well-versed in the DCAT format, so this doesn't throw them off. The target audience for anything related to profiles generally is much broader so we can't count on users having "inside information." For that reason, conforming to basic RDF is more important.

As to your second point, this has been discussed in w3c/dxwg#769 and in particular at this [comment](https://github.com/w3c/dxwg/issues/769#issuecomment-466404671). dcat:Distribution is a class for a "downloadable file", if I understand correctly, and describes the nature of that file. The role defined in PROF is not the nature of the file, it is a relationship between the profile and the file (called 'artifact' in PROF). The same file can have different roles, either in the same profile or in other profiles. So the file description and the role are not one-to-one, and therefore need to be separated. Otherwise, every use of a file with a different role will require a new description, which isn't a good idea, especially when it comes to sharing.

An example of this is the library community that is developing an RDF vocabulary for cataloging, and now has software that allows anyone to create a profile from that same vocabulary based on their own needs and cataloging software. There will be thousands of profiles as libraries using this vocabulary, world-wide, create one or more profiles for their own work. This means that there will be a lot of sharing of the basic vocabulary, some sharing of profiles, and there are also common rules for input that will undoubtedly be "profiled" to fit the many application profiles. The fact that these resources will not be held neatly in silos but will be widely shared means that 1) the descriptors need stable URIs (not bnodes) and 2) the roles and descriptors need to be separate so that they can vary independently.

I really see the DCAT use case and the Profiles use case to be very different, and that is why I don't think that just adopting DCAT's Distribution is a good fit.

**Comment by nicholascar** [comment]
@kcoyle

> ...every use of a file with a different role will require a new description, which isn't a good idea, especially when it comes to sharing.

If you wanted to use a file already used for Profile A with Role X but now for Profile B woth Role Y then yes, you would “require a new description”. The use, and thus the association of that file to a Rrofile - the ResourceDescriptor -  is different.

If you wanted to use a file for Profile A with Role X for Profile B with Role X.... ta da, that’s what <code>isInheritedFrom</code> is for!

If sounds like your would wish to see a register of ResourceDescritors for reuse, perhaps in a community like libraries. Not impossible at all but that, like registers of Profiles themselves need not be handled in PROF but is cataloguing of PROF resources.

**Comment by makxdekkers** [comment]
@nicholascar Can you explain "_that's what `isInheritedFrom` is for_"? Is ResourceDescriptor B inherited from Profile A, or would there just be one ResourceDescriptor that 'is inherited from' both Profile A and B?

**Comment by rob-metalinkage** [comment]
So as far as I can tell we have reconsidered the available alternatives and circled back to the existing model. 
Please review w3c/dxwg#842 to see if that helps any - i think its perhaps an improvement that is closer to familiar words for concept.

we could also consider an entailment rule (can this be done as a owl:propertyChain?)

CONSTRUCT { 
    ?pr dct:format ?f .
} WHERE
{  
  ?pr prof:hasArtifact ?a
   ?a dct:format ?f  
}

**Comment by rob-metalinkage** [comment]
@kcoyle - missed your last comment as i was replying - I do accept your logic about a more general case, so we need to be explicit that the case you put forward is supported. I dont want to throw out the DCAT compatibility however, as we have at least one motivating Use Case to answer for. Hopefully the suggestion above allows us to have our cake an eat it too.  Your argument does convince me we dont want to normatively align with DCAT - it can remain as an optional alignment.

**Comment by rob-metalinkage** [comment]
So - to get back to the issue - is there a concrete proposal for an alternative name we could vote on? 
(i am not averse to making the name more role-centric BTW)
 If not, i suggest we leave it open ion the next PWD and close if we get no better suggestions after comment cycle,

**Comment by nicholascar** [comment]
I've previously mentioned *Profile Distribution* as an informal alignment with DCAT and I like the idea of using *Profile...* as that prevents any other words needing definition like *Resource* etc.

I probably now prefer *Profile Part* as that's even clearer - the thing is part of a *Profile* and we could consider using `dct:hasPart` to indicate it, rather than `prof:hasResource` which seems to do the same thing and reduces the number of new properties.

**Comment by makxdekkers** [comment]
Sounds good to me. Just a minor point would be that it should not be excluded that the Profile Part could contain _all_ constraints the Profile defines.

**Comment by nicholascar** [comment]
@rob-metalinkage what do you make of the concrete proposal to rename the class to _ProfilePart_ with the requirement from @makxdekkers that it be defined to not exclude containing _all_ that the Profile defines?

**Comment by nicholascar** [comment]
The proposal in w3c/dxwg#1061 is to rename `ResourceDescriptor` to `ProfileDistribution` to better conceptually align with the DCAT model (`Profile`/`ProfileDistribution` aligns with `Dataset`/`Distribution` and this also follows the ADMS precedent which uses `Asset`/`AssetDistribution`).

Since continuing discussion is occurring in w3c/dxwg#1061, I'm marking this issue due-for-closing in favour of discussion there.

**Comment by aisaac** [comment]
This is the place where the discussion on naming should happen. So I'm pasting the elements of w3c/dxwg#1061 that are relevant here:

From @rob-metalinkage : My preference has always been for a change to a better name for ResourceDescriptor, and I never much liked hasResource anyway. ProfileObject is really really awful however - every thing in a vocabulary could be called ProfileObject and no interpretation is possible. Given that the prime reason the ResourceDescriptor exists is as a qualified role association a name closer to that might be better. all the other things we want to add to it (as per DCAT distribution), format, conformsTo etc are all clues about its role too that are easier to machine-read than descriptive text ). Could people live with ResourceRole and hasResourceRole ?

From: @nicholascar we already have a ResourceRole class (it was formerly Role but we changes it at 2PWD). How about ProfileDistribution, a la DCAT?

From: @rob-metalinkage : agreed - ResourceRole is already better for the Role descriptor itself so agree best to rename ResourceDescriptor as making it more obvious this akin to DCAT distribution pattern will help, particularly as we have decided not to make profiles dependent on DCAT. We have an informal consensus that the ResourceDescriptor placeholder needs changing.

**Comment by aisaac** [comment]
The proposal to change ResourceDescriptor to ProfileDistribution is tempting, but...
I have done a long study of this and related stuff as part of trying to align our different conceptualization (for this issue and w3c/dxwg#529):
https://docs.google.com/document/d/1uUl8ztXMctnsYTkTH_jV1bA27kqezdHtFRYva18Wsy0/
The relevant part here is "Profile "resource descriptors"/descriptions/specifications/expressions and their encodings/artifacts" especially in "ResourceDescriptor + hasResource"
Since then, the wording of the various PROF elements may have change, but I believe the PROF pattern is still the same. And in that pattern, there is still an issue with aligning to dcat:Distribution, as has been noted here before. Essentially, if `dcat:Distribution`s are serialization- and address-specific, then they fit rather the "artefact" level in PROF. And I certainly DON'T want that one can say that the objects of the `prof:hasArtefact` statements are (DCAT) distributions of a (DCAT) distribution!!!

This is why in the end I prefered `ProfileSpecification` and (even more) `ProfileExpression` as alternatives to `ResourceDescriptor` (and around the Lyon F2F we have also hinted `ProfileComponents` and `ProfileManifestations`)

There is however something that I may have misunderstood about DCAT's Distributions: maybe they are not location-specific? I.e. if a `dcat:Distribution` can have several `dcat:accessURL` or `dcat:downloadURL` then the distributions are a bit more conceptual than what I originally thought, and we can use `ProfileDistribution` for `ResourceRole`. But I'm still skeptical, as I read that DCAT's distributions are serialization specific. To me a same `ResourceDescriptor` in PROF can be available both as a JSON-LD and an RDF/XML artefact representing the same set of RDF statements, and I'm really not sure this fits with DCAT's view on Distribution. I.e. I understand a `dcat:Distribution` can be in JSON-LD or in RDF/XML but not both.

**Comment by smrgeoinfo** [comment]
If we allow that a profile (or any specification) might be considered a kind of dcat:Resource, and if we consider ResourceDescriptor to be a subProperty of [dcat:qualifiedRelation](https://w3c.github.io/dxwg/dcat/#qualified-forms) that associates a Profile with another resource that "defines an aspect - a particular part or feature - of a Profile", the dcat:hadRole on the relationship can specify the nature of the relationship.  The related resource can have many dcat:Distributions to account for different representations or formats of whatever kind of resources it is.  To use this approach, Profile would have to be a sub class of dcat:Resource, and the resources that define aspects of the profile would also have to be dcat:Resources-- a 'prof:ProfileDocument' resource subclass would do nicely, and keep the dcat extensions in the prof namespace.  W3C process considerations aside, wouldn't this solve the problem?

**Comment by aisaac** [comment]
@smrgeoinfo I'm afraid this wouldn't solve the problem. Well it is possible to do it, but then we renounce any ambition to align with dcat:Distribution, one way or another. The qualified relation pattern is a very technical thing, without too much semantics. Well, maybe this is preferable ;-)

But prof:ProfileDocument would again be too specific maybe. And bring too much confusion with the other files/documents that are discussed in our context. As said earlier, `ResourceDescriptor`s aren't specific files, as they can be in different serialization/places.

**Comment by smrgeoinfo** [comment]
Isn't resourceDescriptor actually a relationship between some resource and a Profile resource?  Is the problem that the group is thinking of the target of resourceDescriptor being a particular representation (Distribution), instead of the target of the resourceDescriptor being another resource that has many possible representations (Distributions). This seems entirely compatible with the concept of Distribution, just generalized to recognize that Datasets aren't the only kind of resource that might have various representations (Distributions) available on the web.

**Comment by aisaac** [comment]
@smrgeoinfo yes you are right, the `ResourceDescriptor` was originally though as an association (#638 ). And yes the issue is that some people feel that `ResourceDescriptor`s are akin to dcat:Distribution - which I suspect is not right. In my earlier analysis I had spotted that we were more or less on the same side ;-)

**Comment by smrgeoinfo** [comment]
so here's a proposal:   re-classify resourceDescriptor to be a property with domain prof:Profile and range dcat:Relationship, make prof:Profile a subclass of dcat:Resource, and add a class that represents resources that "define an aspect - a particular part or feature - of a Profile". What to call this last item is open to discussion, but I suggest it could be something generic like 'prof:ProfileResource' .  
![Profile](https://user-images.githubusercontent.com/513380/64991519-e0307480-d886-11e9-82d1-b5d356b0a6f5.png)

**Comment by aisaac** [comment]
@smrgeoinfo sorry but I don't think the pattern solves the issue.
1. I have the same objection to using "resource" here as for any other class or properties. In RDF everything is a resource, so this is very confusing. The pattern also keeps the name resourceDescriptor (the fact that it's on a property doesn't make it more likeable!)
2. dcat:Resource exemplifies the issue of using "resource" in a name, as it's in fact less general than what it sounds. It's defined as "Resource published or curated by a single agent." and "The class of all cataloged resources". These seems at odds actually, it's really two different things lumped in one class. Anyway, whatever these two aspects of dcat:Resource mean, I'm pretty sure they provide no light to the discussion here (the fact of being published by a single agents or added in a catalogue doesn't matter).
3. what's the use of having `ProfileResource` if we have another association class next to it? This makes the pattern even more complex. Many in the WG had already a problem with the fact that there are ResourceDescriptors and artefacts. Your earlier recalling that ResourceDescriptor was an association class would actually be in favour of not introducing another one!

To conclude, I'll re-state that I think the current PROF pattern is actually ok. The issues are the naming of the elements, and the problem of knowing how it aligns with dcat:Distribution (well, in fact as discussed earlier with @smrgeoinfo we may well drop the ambition to align with dcat:Distribution if we can't agree on it...)

**Comment by smrgeoinfo** [comment]
My take on this is that dcat has a nice, extensible pattern relating resources (even if the definition isn't very good), representations of resources, and typed relationships between resources that IMHO can apply very nicely for profiles and documents about profiles. Using a different pattern in prof will be more confusing for users in the long run. Yes, what's in the draft will work, whatever names the classes and properties end up with.  I'll leave it at that.

**Comment by dr-shorthair** [comment]
> dcat:Resource exemplifies the issue of using "resource" in a name

Yes, but that's why it is scoped to `dcat:`. The local name is just the trailing element of a URI. DCAT-Resources are not the same as RDF-Resources.

**Comment by rob-metalinkage** [comment]
Thanks for weighing in - I tjhink @aisaac characterisation is correct - the model is OK but we want to choose the best possible names for things.. 

There is no problem using DCAT for an alternative way of specifying the relation (its an open world assumption in play) - if you do that it might entail under OWL reasoning that a Profile is a dcat:Resource (actually a dcat:Dataset if the resource is a dcat:Distribution which is why formal alignment felt too hard - too much potential argument about what a "Dataset" is !)

So, i think its fine if an implementer chooses to use polymorphism to map the Profile pattern onto the identical DCAT pattern - and an the idea is that an informative alignment is published to do this (again following DCAT precedent exactly).

I dont mind "ProfileResource" at all - we were playing with "ProfileDistribution" - but maybe that makes it too hard to characterise things like guidance notes and mappings to other related models - which may be critical to understand the semantics of a profile. Such things are really not "Distributions" of the Profile at all - they are resources that play a role.

So with all the discussion it seems there are three semantically "valid" choices for the name of the association class proposed so far:
1) prof:ResourceDescriptor
2) prof:ProfileResource
3) prof:Resource

(the latter taking Simon's point that Profile is redundant if scoped to prof: and we just need to get over thinking the same local name has any significance outside the namespace scope! - the problem is that dcat:Resource is what prof:Profile maps to - not what prof:Resource maps to and that causes pain)

I can live with any of these - but not something with a narrower implication about the type of resource or its specific role, as we have mechanisms to handle those aspects in a better way.

i think i lean towards prof:ProfileResource as "Descriptor" is kind of redundant - RDF resources holding metadata about some thing are always descriptors, not the thing..

**Comment by aisaac** [comment]
@dr-shorthair yes I was mildly aware of this before starting my comment, that's why I checked the definition and usage note to be sure of what I was writing. The problem is that not everyone will do this. But well anyway it's not a problem for DCAT!

Btw the fact that I'm relatively ok with it is that `dcat:Resource` is a super-class of other important classes in DCAT (the same way as happens with `prof:Resource` in PROF, another notable use of the R word). It's not harshly counter-intuitive to find something called "resource" when it's a the top of the (local) hierarchy. But that wouldn't be the case if we mint `prof:Resource` or `prof:ProfileResource` instead of `ResourceDescriptor`: even if it is prefixed with `prof`:, my intuition would be that `ProfileResource` would serve as a super-class of (most) of what is in PROF, which is arguably not the case of `ResourceDescriptor`(i.e. it's not a superclass of `prof:Profile`).

**Comment by nicholascar** [comment]
I vote for the options in this order of preference:

1. `prof:ProfileResource`
2. `prof:Resource`
3. `prof:ProfileExpression`
4. `prof:ResourceDescriptor`

**Comment by aisaac** [comment]
I'm wondering why nobody ever reacted to my proposal for "profile expression". Is it such a no-go?

**Comment by nicholascar** [comment]
@aisaac I think I just got a bit lost and missed it Antoine! Too many things on...

I prefer `ProfileExpression` less than the `ProfileResource` though since "expression", to me, reads like an full expression of the (profile) information. So I think `ProfileResource` can more easily be interpreted as _either_ a whole (expression) or a part (expression) of the profile.

I've updated my vote above to include it but, sorry, it's only No. 3!

**Comment by aisaac** [comment]
no worries, @nicholascar, you raise a good point: indeed "expression" could be seen to have a more "complete" meaning than other proposals, and that's a big problem.

How about "component"? It is used in the [DCMI Singapore Framework](https://www.dublincore.org/specifications/dublin-core/singapore-framework/) and seems to have a tad lighter semantic commitment with respect to the "part vs whole" scale (I know "part" was proposed earlier, and didn't like it myself). But maybe the alignment with the DCMI SF is only partial. In fact PROF "components" may have to be more general than DCMI SF (as the components there are presented as "documentation"). But maybe this would still be alright... What do you think @kcoyle @tombaker ?

**Comment by rob-metalinkage** [comment]
guidance is not a component - mappings and examples are not components - but all are releavnt and useful.  Change notes detailing changes between versions etc - all useful resources but hard to fit into more role-specific names.
